doc/0003-wizarding-api.md

```diff
@@ -0,0 +1,227 @@
+# Wizarding API
+
+Date: 2023-07
+
+## Status
+
+Accepted
+
+## Context
+
+Often plugins need a way to display, edit and create SCD elements.
+We want to avoid to put too much effort into implementing the same thing over and over again.
+Also we want to speed up plugin development by extracting some common functionalities out of the plugins.
+
```

Comment:
Is this limited to SCL editing only? or could it e.g. also contain buttons for functions e.g. auto button e.g. Guess content:

![image](https://github.com/openscd/open-scd-core/assets/7427358/04b76f55-6a05-4fd1-9d30-f29f96217c15)

Reply:
Oke, any content will work :)

Reply:
The plug-in author will still be responsible on how the dialog looks?

Reply:
If by plug-in you mean the wizard-library plug-in, then yes. If you mean the plugin that calls the wizard, then no: The idea for the API that we've come up with throughout April 2022 is that the plugin sends a pure declaration of intent: in this case that would be "Please allow the user to create a Substation element in this SCL document!" and then the wizard library would be responsible for displaying some dialog that would allow the user to do that. That dialog, in our default wizard library, would look exactly as you have shown there, with the "Guess content" button. Does that make sense?

Reply:
Same thinking here. The Wizard-Plugins are 100% responsible of their visuals and content.